text/0014-ai-prompt-api.md

```diff
@@ -0,0 +1,174 @@
+# RFC Template
+
+- Start Date: 2025-04-09
+- RFC PR: [electron/rfcs#14](https://github.com/electron/rfcs/pull/14)
+- Electron Issues:
+- Reference Implementation:
+- Status: **Proposed**
+
+# Prompt API support
+
+## Summary
+
+The Prompt API is part of [Built-in AI](https://developer.chrome.com/docs/ai/built-in)
+capabilities in Chromium based browsers. With the Prompt API developers can send
+natural language requests from the renderer for local processing.  This proposal adds
+support for this API in Electron.
+
+## Motivation
+
+Built in AI features are being implemented in Chromium based browsers.  Electron should
+also support these features.
+
+## Guide-level explanation
+
+The Prompt API can be used by Electron app developers by defining a local AI handler
+[utilityProcess](https://www.electronjs.org/docs/latest/api/utility-process) via
+`Session.registerLocalAIHandler` and a new module `localAIHandler`. The utilityProcess
+will allow app developers to proxy Built-in API calls to a local LLM implementation of
+their choice.
+
+For the initial implementation, the local AI handler will only support the prompt API,
+but the long term intention is that this handler would also support other Built-in AI APIs.
+
+### Session
+
+`ses.registerLocalAIHandler(handler)`
+* `handler` [`UtilityProcess`](utility-process.md#class-utilityprocess)
+  
+Returns `string` - The ID of the registered local AI handler.
+
+`ses.unregisterLocalAIHandlerScript(id)`
+
+* `id` string - local AI handler ID
+
+Unregisters the specified local AI handler utilityProcess.
+
+_NOTE: The app developer will be responsible for managing the lifetime of
+the utilityProcess; this method merely handles disassociation of the
+utilityProcess as an AI handler.
```

Comment:
Maybe we could opt for `ses.registerLocalAIHandler(null)` now that scripts are no longer used?

Reply:
Yea, that might be cleaner, assuming we only want to allow a single handler utility process per session. That would also remove the need for the returned ID.

Reply:
The intention was to allow for multiple handlers but maybe it is fine to just have one.